packages/main/src/StepInput.ts

```diff
@@ -512,17 +546,25 @@ class StepInput extends UI5Element implements IFormElement {
 		}
 	}
 
+	_isPrecisionValid() {
+		// gets either "." or "," as delimiter which is based on locale, and splits the number by it
+		const delimiter = !Number.isInteger(Number(this.input.value)) && this.input.value.includes(".") ? "." : ",";
+		const numberParts = this.input.value.split(delimiter);
+		const decimalPartLength = numberParts.length > 1 ? numberParts[1].length : 0;
+		const isPrecisionValid = decimalPartLength === this.valuePrecision;
+
+		return isPrecisionValid;
+	}
+
 	_onInputChange() {
 		if (this.input.value === "") {
-			this.input.value = (this.min || 0) as unknown as string;
-		}
-		const inputValue = this._preciseValue(parseFloat(this.input.value));
-		if (this.value !== this._previousValue || this.value !== inputValue) {
-			this.value = inputValue;
-			this._validate();
-			this._setButtonState();
-			this._fireChangeEvent();
+			this.input.value = (this.min || 0).toFixed(this.valuePrecision);
 		}
+		const inputValue = Number(this.input.value);
```

Comment:
We don't need to check anymore if the value is changed before firing the change event?

Reply:
I've got mistaken by the fact that, since now we notify the end user for the input validity by updating `value-state`, that we now should always fire change event, to check value validity and update the value state. However it is conceptionally wrong to fire change event even if the value isn't updated. So I would have to fix that. Thank you.